src/core/structures/annotation/PDFAnnotation.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  PDFArray,
+  PDFDict,
+  PDFName,
+  PDFNumber,
+  PDFRectangle,
+  PDFString,
+} from 'src/core';
+
+const annotationTypes = [
+  PDFName.Text,
+  PDFName.Link,
+  PDFName.FreeText,
+  PDFName.Line,
+  PDFName.Square,
+  PDFName.Circle,
+  PDFName.Polygon,
+  PDFName.PolyLine,
+  PDFName.Highlight,
+  PDFName.Underline,
+  PDFName.Squiggly,
+  PDFName.StrikeOut,
+  PDFName.Stamp,
+  PDFName.Caret,
+  PDFName.Ink,
+  PDFName.Popup,
+  PDFName.FileAttachment,
+  PDFName.Sound,
+  PDFName.Movie,
+  PDFName.Widget,
+  PDFName.Screen,
+  PDFName.PrinterMark,
+  PDFName.TrapNet,
+  PDFName.Watermark,
+  PDFName.ThreeD,
+  PDFName.Redact,
+];
+
+class PDFAnnotation {
+  static fromDict(dict: PDFDict): PDFAnnotation {
+    return new PDFAnnotation(dict);
+  }
+
+  readonly dict: PDFDict;
+
+  protected constructor(dict: PDFDict) {
+    this.dict = dict;
+    if (!annotationTypes.includes(this.Subtype())) {
+      throw new Error('Invalid annotation type');
+    }
+  }
+
+  Type(): PDFName | undefined {
+    return this.dict.lookupMaybe(PDFName.Type, PDFName);
+  }
+
+  Subtype(): PDFName {
+    return this.dict.lookup(PDFName.Subtype, PDFName);
+  }
+
+  Rect(): PDFRectangle {
```

Comment:
`PDFRectangle` is not a subclass of `PDFObject`, so returning it here violates our convention of only returning primitives that can be stored in the `PDFContext` from the low-level getters. This method should just return a `PDFArray`, and another helper could be created for returning a `PDFRectangle` structure.

Reply:
Updated